analyzers/src/SonarAnalyzer.CSharp/Rules/NewGuidShouldNotBeUsed.cs

```diff
@@ -25,35 +25,63 @@
 
         protected override void Initialize(SonarAnalysisContext context)
         {
-            context.RegisterSyntaxNodeActionInNonGenerated(
-                c =>
+            DetectIssueInConstructors(context);
+            DetectIssueInDefaultExpressions(context);
+        }
+
+        private static void DetectIssueInConstructors(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var objectCreationSyntax = ObjectCreationFactory.Create(c.Node);
+                if (objectCreationSyntax.ArgumentList?.Arguments.Count == 0
+                    && objectCreationSyntax.MethodSymbol(c.SemanticModel) is IMethodSymbol methodSymbol
+                    && methodSymbol.ContainingType.Is(KnownType.System_Guid))
+                {
+                    c.ReportIssue(Diagnostic.Create(Rule, objectCreationSyntax.Expression.GetLocation()));
+                }
+            },
+            SyntaxKind.ObjectCreationExpression,
+            SyntaxKindEx.ImplicitObjectCreationExpression);
+
+        private static void DetectIssueInDefaultExpressions(SonarAnalysisContext context) =>
+            context.RegisterSyntaxNodeActionInNonGenerated(c =>
+            {
+                var expressionSyntax = (ExpressionSyntax)c.Node;
+                if (expressionSyntax.IsKind(SyntaxKindEx.DefaultLiteralExpression)
+                    && c.SemanticModel.GetTypeInfo(expressionSyntax).ConvertedType.Is(KnownType.System_Guid))
+                {
+                    c.ReportIssue(Diagnostic.Create(Rule, expressionSyntax.GetLocation()));
+                }
+                else if (expressionSyntax.IsKind(SyntaxKind.DefaultExpression)
+                         && DefaultExpressionIdentifierIsGuid((DefaultExpressionSyntax)expressionSyntax))
                 {
-                    var objectCreationSyntax = (ObjectCreationExpressionSyntax)c.Node;
+                    c.ReportIssue(Diagnostic.Create(Rule, expressionSyntax.GetLocation()));
+                }
+            },
+            SyntaxKind.DefaultExpression,
+            SyntaxKindEx.DefaultLiteralExpression);
 
-                    if (objectCreationSyntax.ArgumentList?.Arguments.Count == 0 &&
-                        c.SemanticModel.GetSymbolInfo(objectCreationSyntax).Symbol is IMethodSymbol methodSymbol &&
-                        methodSymbol.ContainingType.Is(KnownType.System_Guid))
-                    {
-                        c.ReportIssue(Diagnostic.Create(rule, objectCreationSyntax.GetLocation()));
-                    }
-                },
-                SyntaxKind.ObjectCreationExpression);
+        private static bool DefaultExpressionIdentifierIsGuid(DefaultExpressionSyntax defaultExpression)
+        {
+            var typeName = defaultExpression.Type.ToString();
+            return typeName == "Guid" || typeName == "System.Guid";
         }
```

Comment:
I've realized now that it can be reduced to:
```
        private static bool DefaultExpressionIdentifierIsGuid(DefaultExpressionSyntax defaultExpression) =>
            defaultExpression.Type.ToString() is var typeName
            && (typeName == "Guid" || typeName == "System.Guid");
```

Up to you if you want to change it.